services/opt/service.go

```diff
@@ -0,0 +1,68 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package opt
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+
+	"github.com/containerd/containerd/plugin"
+	"github.com/pkg/errors"
+)
+
+// Config for the opt manager
+type Config struct {
+	// Path for the opt directory
+	Path string `toml:"path"`
+}
+
+func init() {
+	plugin.Register(&plugin.Registration{
+		Type: plugin.InternalPlugin,
+		ID:   "opt",
```

Comment:
Ok, so we now have an opt plugin to allow introspection of the opt location for binary resolution. Is that how this is working?

Reply:
yup, it seems like a cleaner way to provide configuration from the daemon to clients

Reply:
and it allows you to disable it in the config as well.  our plugin setup is nice